src/env.rs

```diff
@@ -456,4 +456,84 @@ mod tests {
         writer.put("foo", &Value::I64(1235)).expect("wrote");
         writer.delete_value("foo", &Value::I64(1234)).expect("deleted");
     }
+
+    #[test]
+    fn test_iter() {
+        let root = TempDir::new("test_iter").expect("tempdir");
+        fs::create_dir_all(root.path()).expect("dir created");
+        let k = Rkv::new(root.path()).expect("new succeeded");
+        let sk: Store<&str> = k.create_or_open("sk").expect("opened");
+
+        // An iterator over an empty store returns no values.
+        {
+            let reader = sk.read(&k).unwrap();
+            let mut iter = reader.iter().unwrap();
+            assert!(iter.next().is_none());
+        }
+
+        let mut writer = sk.write(&k).expect("writer");
+        writer.put("foo", &Value::I64(1234)).expect("wrote");
+        writer.put("noo", &Value::F64(1234.0.into())).expect("wrote");
+        writer.put("bar", &Value::Bool(true)).expect("wrote");
+        writer.put("baz", &Value::Str("héllo, yöu")).expect("wrote");
+        writer.commit().expect("committed");
+
+        let reader = sk.read(&k).unwrap();
+
+        // Reader.iter() returns (key, value) tuples ordered by key.
+        let mut iter = reader.iter().unwrap();
+        let (key, val) = iter.next().unwrap();
+        assert_eq!(str::from_utf8(key).expect("key"), "bar");
+        assert_eq!(val.expect("value"), Some(Value::Bool(true)));
+        let (key, val) = iter.next().unwrap();
+        assert_eq!(str::from_utf8(key).expect("key"), "baz");
+        assert_eq!(val.expect("value"), Some(Value::Str("héllo, yöu")));
+        let (key, val) = iter.next().unwrap();
+        assert_eq!(str::from_utf8(key).expect("key"), "foo");
+        assert_eq!(val.expect("value"), Some(Value::I64(1234)));
+        let (key, val) = iter.next().unwrap();
+        assert_eq!(str::from_utf8(key).expect("key"), "noo");
+        assert_eq!(val.expect("value"), Some(Value::F64(1234.0.into())));
+        assert!(iter.next().is_none());
+
+        // Iterators don't loop.  Once one returns None, additional calls
+        // to its next() method will always return None.
+        assert!(iter.next().is_none());
+
+        // Reader.iter_from() begins iteration at the first key equal to
+        // or greater than the given key.  In this case, the first (and only)
+        // key greater than "moo" is "noo", so that's the only result.
```

Comment:
Can you copypaste this test to include retrieval by prefix (e.g., `"no"`)?

Reply:
Yes, done in fba629f.